src/mame/machine/nb1412m2.cpp

```diff
@@ -290,8 +290,11 @@ void nb1412m2_device::timer_w(uint8_t data)
 	if(data != 1)
 		logerror("nb1412m2: timer_w with data == %02x\n",data);
 
-	// TODO: timing of this, related to m_const90?
-	m_timer->adjust(attotime::from_hz(960));
+	// The DAC frequency and timer clock are linked.
+	// When changes the DAC clock, Sound driver set wait loop count ($C010)
+	// in the range of 2 to 4 in order to keep the tempo of BGM even if changed clock.
+	// This value verified with BGM tempo of PCB.
+	m_timer->adjust(attotime::from_hz(446 * m_timer_rate));
```

Comment:
This has a large prime factor (223).  It’s somewhat unusual to have large prime factors, as it requires more logic.  What’s actually generating the delay?

Reply:
I agree with you.
It is the result of searching for a value that matches BGM tempo, but it is possible that there is a slight error.
I will try to find a more optimal value.

Reply:
When I compared the name entry screen again, it seems that 446 was only slightly slower.
I changed it to 450, which seems to have solved both the prime number problem and slowness.